client/src/reducers/undoableConfig.js

```diff
@@ -42,7 +43,9 @@ Entire action must be identical (all keys).
 const debounceOnActions = new Set([
   "color by categorical metadata",
   "color by continuous metadata",
-  "color by expression"
+  "color by expression",
+
+  "show centroid labels for category"
```

Comment:
why are you debouncing this?  I would have assumed you would _not_ want it debounced.